src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/configuration/MultiThreadedConfig.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.configuration;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.consumer.MessageConsumer;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.service.LoopingMessageProcessor;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.service.impl.LoopingMessageProcessorServiceImpl;
+
+import java.util.concurrent.Executor;
+
+@Configuration
+public class MultiThreadedConfig {
+
+    @Bean("main-looping-consumer")
+    LoopingMessageProcessor mainLoopingConsumer(ApplicationLogger logger,
+                                                @Qualifier("incoming-message-consumer") MessageConsumer messageConsumer
+    ) {
+        return new LoopingMessageProcessorServiceImpl(messageConsumer, logger, "main-looping-consumer");
+    }
+
+    @Bean("retry-looping-consumer")
+    LoopingMessageProcessor retryLoopingConsumer(ApplicationLogger logger,
+                                                 @Qualifier("retry-message-consumer") MessageConsumer messageConsumer
+    ) {
+        return new LoopingMessageProcessorServiceImpl(messageConsumer, logger, "retry-looping-consumer");
+    }
+
+    @Bean
+    Executor taskExecutor() {
```

Comment:
I'd create a unit test for this method, I could set poolsize to 0, set thread prefix to "extensionsApp" and the tests would still pass

Reply:
I have thought about that, but the Executor object returned doesn't have any way of asserting what the values are.

Reply:
ah ok

Reply:
It's because the returned type is the interface. If we cast the object back to ThreadPoolExecutor in the test we can check the values:
` MultiThreadedConfig config = new MultiThreadedConfig();
   ThreadPoolTaskExecutor executor = (ThreadPoolTaskExecutor) config.taskExecutor();
   executor.getMaxPoolSize()`